lib/src/map_widget.dart

```diff
@@ -243,39 +208,43 @@ class _MapWidgetState extends State<MapWidget> {
       final controller = await _controller.future;
       controller.dispose();
     }
+    _suffixesRegistry.releaseSuffix(_suffix);
+    _events.dispose();
   }
 
   @override
   void didUpdateWidget(MapWidget oldWidget) {
     super.didUpdateWidget(oldWidget);
+
+    _updateSubscriptions();
+  }
+
+  void _updateSubscriptions() {
+    _events.updateSubscriptions(
+        onStyleLoadedListener: widget.onStyleLoadedListener,
+        onCameraChangeListener: widget.onCameraChangeListener,
+        onMapIdleListener: widget.onMapIdleListener,
+        onMapLoadedListener: widget.onMapLoadedListener,
+        onMapLoadErrorListener: widget.onMapLoadErrorListener,
+        onRenderFrameFinishedListener: widget.onRenderFrameFinishedListener,
+        onRenderFrameStartedListener: widget.onRenderFrameStartedListener,
+        onSourceAddedListener: widget.onSourceAddedListener,
+        onSourceDataLoadedListener: widget.onSourceDataLoadedListener,
+        onSourceRemovedListener: widget.onSourceRemovedListener,
+        onStyleDataLoadedListener: widget.onStyleDataLoadedListener,
+        onStyleImageMissingListener: widget.onStyleImageMissingListener,
+        onStyleImageUnusedListener: widget.onStyleImageUnusedListener,
+        onResourceRequestListener: widget.onResourceRequestListener);
   }
 
   void onPlatformViewCreated(int id) {
-    _mapboxMapsPlatform.initPlatform();
-    final MapboxMap controller = MapboxMap(
-      mapboxMapsPlatform: _mapboxMapsPlatform,
-      onStyleLoadedListener: widget.onStyleLoadedListener,
-      onCameraChangeListener: widget.onCameraChangeListener,
-      onMapIdleListener: widget.onMapIdleListener,
-      onMapLoadedListener: widget.onMapLoadedListener,
-      onMapLoadErrorListener: widget.onMapLoadErrorListener,
-      onRenderFrameFinishedListener: widget.onRenderFrameFinishedListener,
-      onRenderFrameStartedListener: widget.onRenderFrameStartedListener,
-      onSourceAddedListener: widget.onSourceAddedListener,
-      onSourceDataLoadedListener: widget.onSourceDataLoadedListener,
-      onSourceRemovedListener: widget.onSourceRemovedListener,
-      onStyleDataLoadedListener: widget.onStyleDataLoadedListener,
-      onStyleImageMissingListener: widget.onStyleImageMissingListener,
-      onStyleImageUnusedListener: widget.onStyleImageUnusedListener,
-      onResourceRequestListener: widget.onResourceRequestListener,
-      onMapTapListener: widget.onTapListener,
```

Comment:
`onMapTapListener` does no longer works in our app after this commit. Should we migrate to something else. If so what is it?

Reply:
Thanks for the heads up, this was removed accidentally, it is restored in https://github.com/mapbox/mapbox-maps-flutter/pull/526